enhancements/ingress/haproxy-healthcheckinterval.md

```diff
@@ -0,0 +1,301 @@
+check interval. Performance validation is not in scope for this proposal.
+
+Other parameters may come into play for adjusting the healthcheck interval, but will not be covered in this proposal.
+It is worth checking in the future, for the impact of `spread-checks`, `max-spread-checks`, `fastinter`, `downinter`,
+and the use of passive health checks rather than active health checks.
+
+## Proposal
+
+In the HAProxy configuration manual, there are three settings that work together to determine the backend healthcheck
+interval:
+- `timeout connect <duration>`: maximum time to wait for a connection attempt to a server to succeed
+- `timeout check <duration>`: sets additional check timeout, but only after a connection has already been established
+- `check inter <duration>`: sets the interval between two consective healthchecks; serves as a timeout for backend
+  healthchecks if `timeout check` is not set
+
+We currently set all of these parameters, in the following way:
+
+| Name | Scope | Default |
+| ---------------|-------- |-----------|
+| timeout check | backend | 5s |
+| timeout connect | default | 5s |
+| check inter| backend | 5s
+
+Because `timeout connect` is currently set at the default scope, it applies to all connection attempts, even healthcheck probes.
+
+`timeout check` is currently set for plain HTTP backends, or backends with TLS termination set to edge, re-encrypt, or
+passthrough.  
+
+`check inter` is set per server for the same backends as `timeout check`, with some additional constraints.  For plain
+HTTP backends, or backends with TLS termination set to edge, re-encrypt, or passthrough, it is set only if the
+endpoint has not disabled healthchecks and the configuration has more than one active endpoint.  With a dynamic
+configuration manager, it is set only for re-encrypt routes, for dynamic servers. The duration may come from (evaluated in
+order), the `router.openshift.io/haproxy.health.check.interval` annotation on the route, a (currently hidden)
+ROUTER_BACKEND_CHECK_INTERVAL environment variable, or 5 seconds as a default if the previous two values are unset.
+
+Whether `timeout check` is set has an impact on the `timeout connect` for healthchecks, as [documented in the HAProxy
+configuration manual](http://cbonte.github.io/haproxy-dconv/2.2/configuration.html#4.2-timeout%20check):
+- if `timeout check` is set, HAProxy "uses min(`timeout connect`, `inter`) as a connect timeout for check and `timeout check`
+as an additional read timeout"
+- if `timeout check` is not set, HAProxy "uses `inter` for the complete check timeout (connect + read)"
+
+This proposal will add a new optional parameter, `HealthCheckInterval` to the `IngressControllerTuningOptions`, to set
+the duration used in `check inter` at the defaults scope.  By setting a specific `HealthCheckInterval`, this will also
+remove the setting for `timeout check` at the backend scope so that `timeout check` won't impact the healthcheck interval.
```

Comment:
No, I meant that if `HealthCheckInterval` is set, it would remove the `timeout check` at the backend scope.  The hardcoded `timeout check` would be conditionally set in the haproxy config template.